NorthstarDedicatedTest/serverauthentication.cpp

```diff
@@ -141,6 +144,28 @@ void ServerAuthenticationManager::StopPlayerAuthServer()
 	m_playerAuthServer.stop();
 }
 
+char* ServerAuthenticationManager::VerifyPlayerName(void* player, char* authToken, char* name)
+{
+	std::lock_guard<std::mutex> guard(m_authDataMutex);
+
+	if (!m_authData.empty() && m_authData.count(std::string(authToken)))
+	{
+		AuthData authData = m_authData[authToken];
+
+		bool nameAccepted = (!*authData.username || !strcmp(name, authData.username));
+
+		if (!nameAccepted && g_MasterServerManager->m_bRequireClientAuth && !CVar_ns_auth_allow_insecure->m_nValue)
+		{
+			strcpy(name, authData.username);
```

Comment:
Use strncpy and leave a comment explaining the 64-character limit. This will reduce the risk of regressions being introduced if the auth code is refactored.